dev-python/slimit/slimit-0.8.1.ebuild

```diff
@@ -0,0 +1,30 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 pypy{,3} python3_{4,5,6} )
+inherit distutils-r1
+
+DESCRIPTION="A JavaScript minifier written in Python"
+HOMEPAGE="https://slimit.readthedocs.io/en/latest/"
+SRC_URI="https://github.com/rspivak/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="dev-python/ply[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]"
+DEPEND="${RDEPEND}
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytest-runner[${PYTHON_USEDEP}]
+	)"
+
+PATCHES=( "${FILESDIR}/${P}-fix-python3.patch" )
```

Comment:
Do you still need this huge patch? If it's just about older versions of Python 3, just remove those versions. There's no point in supporting them in new packages.

Reply:
No, it's still necessary for python3.6 and pyp3, otherwise:
```
============================================================= FAILURES =============================================================
____________________________________________________ LexerTestCase.test_case_1 _____________________________________________________

self = <slimit.tests.test_lexer.LexerTestCase testMethod=test_case_1>

    def test_func(self):
        lexer = self._get_lexer()
        lexer.input(input)
        result = ['%s %s' % (token.type, token.value) for token in lexer]
>       self.assertListEqual(result, expected)

../slimit-0.8.1-python3_6/lib/slimit/tests/test_lexer.py:278: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../slimit-0.8.1-python3_6/lib/slimit/tests/test_lexer.py:55: in assertListEqual
    self.fail('Lists differ:\n' + message)
E   AssertionError: Lists differ:
E   - ['ID u03c0', 'ID u03c0_tail', 'VAR var', 'ID ua67c']
E   ?                                ^^^    ^^^^^^^
E   
E   + ['ID \\u03c0', 'ID \\u03c0_tail', 'ID var\\ua67c']
E   ?      ++            ++              ^^    ^^
------------------------------------------------------- Captured stdout call -------------------------------------------------------
Illegal character '\\' at 1:0 after None
Illegal character '\\' at 1:7 after LexToken(ID,'u03c0',1,1)
Illegal character '\\' at 1:22 after LexToken(VAR,'var',1,19)
```